autosubmit/git/autosubmit_git.py

```diff
@@ -257,3 +259,31 @@ def clone_repository(as_conf, force, hpcarch):
             shutil.rmtree(project_backup_path)
 
         return True
+
+    @staticmethod
+    def check_unpushed_changes(expid: str) -> None:
+        """
+        Raises an AutosubmitCritical error if the experiment is operational, the platform is Git, and there are unpushed changes.
+
+        Args: expid (str): The experiment ID.
+
+        Returns: None
+        """
+        if expid[0] == 'o':
+            origin = Path(BasicConfig.expid_dir(expid).joinpath("conf/expdef_{}.yml".format(expid)))
```

Comment:
This file may not exist, as not all AS4 experiments have `expdef_...` files. It would have been better, I think, if we used the parsed configuration, ignoring the file names, to retrieve whether we have a Git project or not. I think ClimateDT experiments may be using that file now, but for users it should be optional -- cc @TracyMcBean